README.md

```diff
@@ -662,6 +664,23 @@ type SomeObject = NumberDictionary<{ a: number; b: string }>;
 type EmptyObject = NumberDictionary<{}>;
 ```
 
+### NonEmptyArray
+
+Useful for accepting only arrays containing at least one element.
+
+```typescript
+type NonEmptyRestParameters = NonEmptyArray<number>;
+// Result: [number, ...number[]]
+
+// declare function expression type accepting some rest parameters, but at least one element for the rest parameters is required
+type FunctionAcceptingRestParameters = (someString: string, ...args: NonEmptyRestParameters<number>) => void;
+
+// declare some non-empty array variables
+const okay: NonEmptyArray<number> = [1, 2];
+const alsoOkay: NonEmptyArray<number> = [1];
+const error: NonEmptyArray<number> = []; // Type '[]' is not assignable to type 'NonEmptyArray<number>'. Source has 0 element(s) but target requires 1.
```

Comment:
Might be worth to have 2 lines instead of 1:

```suggestion
// @ts-expect-error: Type '[]' is not assignable to type 'NonEmptyArray<number>'. Source has 0 element(s) but target requires 1.
const error: NonEmptyArray<number> = [];
```

WDYT?